mlflow/store/file_store.py

```diff
@@ -59,8 +59,8 @@ def __init__(self, root_directory=None, artifact_root_uri=None):
         Create a new FileStore with the given root directory and a given default artifact root URI.
         """
         super(FileStore, self).__init__()
-        self.root_directory = root_directory or _default_root_dir()
-        self.artifact_root_uri = artifact_root_uri or self.root_directory
+        self.root_directory = parse_path(root_directory or _default_root_dir())
+        self.artifact_root_uri = artifact_root_uri or "file://" + self.root_directory
```

Comment:
I can make this os dependent, otherwise file:/// + an absolute_path in linux will result in file://// which is unsupported. file:// supports both with the netloc + path approach but as you mentioned is not a valid uri.

I think the simplest correct uri would be to prepend file:/// and remove leading / from self.root_directory. Do you agree?

Reply:
Ah, makes sense!

Wouldn't removing '/' from the self.root_directory make the path relative? Or did you mean you would remove the '/' only for the artifact_uri? I agree the safest is to get the uri into a standardized format starting with file:///.

Reply:
file:/// makes all paths absolute. file:// support relative paths by combining netloc  + path. The start of the relative path is always in netloc

urllib.parse.urlparse("file://path/to/my/stuff")
ParseResult(scheme='file', netloc='path', path='/to/my/stuff', params='', query='', fragment='')

urllib.parse.urlparse("file:///path/to/my/stuff")
ParseResult(scheme='file', netloc='', path='/path/to/my/stuff', params='', query='', fragment='')

Reply:
If I understand it correctly, file:///... uri should always be an absolute path? 
So I think we should always make the path absolute when creating the uri and make sure it has file:/// prefix.

Reply:
This makes sense. I will make the changes and simplify parse_path as well.